src/golang/cmd/server/handler/v2/node_check_results_get.go

```diff
@@ -0,0 +1,147 @@
+// Response:
+//	Body:
+//		`[]response.OperatorWithArtifactResultNode`
+
+type NodeCheckResultsGetHandler struct {
+	nodeGetHandler
+	handler.GetHandler
+
+	Database database.Database
+
+	WorkflowRepo       repos.Workflow
+	DAGRepo            repos.DAG
+	OperatorRepo       repos.Operator
+	OperatorResultRepo repos.OperatorResult
+	ArtifactRepo       repos.Artifact
+	ArtifactResultRepo repos.ArtifactResult
+}
+
+func (*NodeCheckResultsGetHandler) Name() string {
+	return "NodeCheckResultsGet"
+}
+
+func (h *NodeCheckResultsGetHandler) Prepare(r *http.Request) (interface{}, int, error) {
+	return h.nodeGetHandler.Prepare(r)
+}
+
+func (h *NodeCheckResultsGetHandler) Perform(ctx context.Context, interfaceArgs interface{}) (interface{}, int, error) {
+	args := interfaceArgs.(*nodeGetArgs)
+
+	artfID := args.nodeID
+	wfID := args.workflowID
+
+	emptyResponse := []response.OperatorWithArtifactResultNode{}
+
+	dbOperatorWithArtifactNodes, err := h.OperatorRepo.GetOperatorWithArtifactByArtifactIdNodeBatch(ctx, []uuid.UUID{artfID}, h.Database)
+	if err != nil {
+		return nil, http.StatusInternalServerError, errors.Wrap(err, "Unexpected error reading check node.")
+	}
+	dbOperatorWithArtifactNode := views.OperatorWithArtifactNode{}
+	if len(dbOperatorWithArtifactNodes) == 0 {
+		return emptyResponse, http.StatusOK, nil
+	} else {
+		dbOperatorWithArtifactNode = dbOperatorWithArtifactNodes[0]
+	}
+
+	results, err := h.OperatorResultRepo.GetOperatorWithArtifactResultNodesByOperatorNameAndWorkflow(ctx, dbOperatorWithArtifactNode.Name, wfID, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to retrieve check results.")
+	}
+
+	if len(results) == 0 {
+		return emptyResponse, http.StatusOK, nil
+	}
+
+	resultArtifactIds := make([]uuid.UUID, 0, len(results))
+	for _, result := range results {
+		resultArtifactIds = append(resultArtifactIds, result.ArtifactResultID)
+	}
+
+	artfResultToDAG, err := h.DAGRepo.GetByArtifactResultBatch(ctx, resultArtifactIds, h.Database)
+	if err != nil {
+		return emptyResponse, http.StatusInternalServerError, errors.Wrap(err, "Unable to retrieve workflow dags.")
+	}
+
+	// maps from db dag Ids
+	dbDagByDagId := make(map[uuid.UUID]models.DAG, len(artfResultToDAG))
+	nodeResultByDagId := make(map[uuid.UUID][]views.OperatorWithArtifactResultNode, len(artfResultToDAG))
+	for _, nodeResult := range results {
+		if dbDag, ok := artfResultToDAG[nodeResult.ArtifactResultID]; ok {
+			if _, okDagsMap := dbDagByDagId[dbDag.ID]; !okDagsMap {
+				dbDagByDagId[dbDag.ID] = dbDag
+			}
+
+			nodeResultByDagId[dbDag.ID] = append(nodeResultByDagId[dbDag.ID], nodeResult)
+		} else {
+			return emptyResponse, http.StatusInternalServerError, errors.Newf("Error retrieving dag associated with artifact result %s", nodeResult.ArtifactResultID)
+		}
+	}
+
+	responses := make([]response.OperatorWithArtifactResultNode, 0, len(results))
+	for dbDagId, nodeResults := range nodeResultByDagId {
```

Comment:
Try to consolidate these repeating parts. We can file a task if the solution is not immediately clear